sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,416 @@
+"""GaussianMixtureIC"""
+
+# Author: Thomas Athey <tathey1@jhmi.edu>
+# Modified by: Benjamin Pedigo <bpedigo@jhu.edu>
+#              Tingshan Liu <tliu68@jhmi.edu>
+
+
+import numpy as np
+import joblib
+from ..utils.fixes import parse_version
+from ..utils.parallel import delayed, Parallel
+from ..utils import check_scalar
+from ..utils.validation import check_is_fitted, check_random_state
+
+from . import GaussianMixture
+from ..base import BaseEstimator, ClusterMixin
+from ..model_selection import ParameterGrid
+
+
+
+
+class GaussianMixtureIC(ClusterMixin, BaseEstimator):
+    """Gaussian mixture with BIC/AIC.
+
+    Automatic Gaussian Mixture Model (GMM) selection via the
+    Bayesian Information Criterion (BIC)
+    or the Akaike Information Criterion (AIC).
+
+    Different combinations of initialization, GMM,
+    and cluster numbers are used and the clustering
+    with the best selection criterion (BIC or AIC) is chosen.
```

Comment:
suggest making this match LassoLarsIC a bit closer, eg "Such criteria are useful to select the value of the regularization parameter by making a trade-off between the goodness of fit and the complexity of the model." could basically replace "regularization parameter" with "gaussian mixture parameters"